src/scheduler/job_info.c

```diff
@@ -3426,11 +3429,42 @@ find_jobs_to_preempt(status *policy, resource_resv *hjob, server_info *sinfo, in
 
 		update_universe_on_end(npolicy, pjob,  "S", NO_ALLPART);
 		rjobs_count--;
+		/* Check if any of the previously preempted job increased its preemption priority */
+		for (ind = 0; pjobs[ind] != NULL; ind++) {
+			if (pjobs[ind]->job->preempt >= nhjob->job->preempt) {
+				dont_preempt_job = 1;
+				break;
+			}
+		}
+		/* Check if the job we just ended increases its priority. If so, don't preempt this job */
+		if (dont_preempt_job || pjob->job->preempt >= nhjob->job->preempt) {
+			schd_error *tmp_err = new_schd_error();
+			clear_schd_error(tmp_err);
+			remove_resresv_from_array(rjobs_subset, pjob);
+			log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_INFO, pjob->name,
+				  "Preempting job will escalate its priority or priority of other jobs, not preempting it");
```

Comment:
Sorry for the naive question :) but why would preempting one job increase the preemption priority of some other job?

Reply:
One of the comment above covers this with an example.